src/stages/glue-transform.ts

```diff
@@ -0,0 +1,68 @@
+import * as glueAlpha from '@aws-cdk/aws-glue-alpha';
+import * as glue from 'aws-cdk-lib/aws-glue';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { Construct } from 'constructs';
+import { StateMachineStage, StateMachineStageProps } from '../pipelines/stage';
+
+export interface GlueTransformStageProps extends StateMachineStageProps {
+  readonly job?: glueAlpha.Job;
+  readonly jobName: string;
+  readonly jobArgs?: { [key: string]: any };
+  readonly crawlerName?: string;
+  readonly cfnCrawler?: glue.CfnCrawler;
+}
+
+export class GlueTransformStage extends StateMachineStage {
+  private readonly job?: glueAlpha.Job;
+  private readonly jobName?: string;
+  private readonly jobArgs?: { [key: string]: any };
+  private readonly crawlerName?: string;
+  private readonly cfnCrawler?: glue.CfnCrawler;
+
+  constructor(scope: Construct, id: string, props: GlueTransformStageProps) {
+    super(scope, id, props);
+
+    this.job = props.job;
+    this.jobName = props.jobName;
+    this.jobArgs = props.jobArgs;
+    this.crawlerName = props.crawlerName;
+    this.cfnCrawler = props.cfnCrawler;
+
+    if (this.jobName == undefined && this.job == undefined) {
```

Comment:
I think we could simpligy this as
```typescript
if (!this.jobName && this.job) {
  throw ...
}
```

This will have the side effect of also throwing errors if an empty string is provided as a job name for some reason.

Reply:
Doesn't it have to be `if (!(this.jobName && this.job))` or ``if (!this.jobName && !this.job)` ? @LeonLuttenberger

Reply:
Thanks guys